mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/uwazi/entry/UwaziParser.kt

```diff
@@ -202,20 +210,38 @@ class UwaziParser(private val context: Context?) {
                 widgetMediaFiles.add(answer)
             }
         }
+        template?.entityRow?.properties?.filter { property -> property.type == UWAZI_DATATYPE_RELATIONSHIP }
+            ?.forEach { property ->
+                val selectedEntities: ArrayList<Any> = ArrayList()
+                uwaziFormView.entities.forEach { selectedEntity ->
+                    property.entities?.filter { it.id == selectedEntity.value }?.forEach {
+                        selectedEntities.add(UwaziRelationShipEntity(it.id, it.label))
+                    }
+                }
+                property.selectedEntities = selectedEntities as List<UwaziRelationShipEntity>
+                answers.filter { answer -> answer.key == property.name }.forEach { answer ->
+                    hashmap[answer.key] = selectedEntities as List<UwaziRelationShipEntity>
+                }
+            }
+        val entities = ArrayList<UwaziRelationShipEntity>()
+        for (answer in uwaziFormView.entities) {
+            if (answer != null) {
+                entities.add(answer)
+            }
+        }
```

Comment:
why we don't  use stream operator as we are using java +8 it would make the code much more clearer ...